src/mjolnir/landmark_database_builder.cc

```diff
@@ -0,0 +1,254 @@
+#include <cstdint>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "baldr/graphconstants.h"
+#include "filesystem.h"
+#include "mjolnir/landmark_database_builder.h"
+#include "mjolnir/util.h"
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/geometries.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+#include <geos_c.h>
+
+namespace valhalla {
+namespace mjolnir {
+bool LandmarkDatabase::connect_database() {
+  switch (access_mode_) {
+    case AccessMode::ReadWriteCreate:
+      open_flags = SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE;
+      break;
+    case AccessMode::ReadOnly:
+      open_flags = SQLITE_OPEN_READONLY;
+      break;
+    case AccessMode::ReadWrite:
+      open_flags = SQLITE_OPEN_READWRITE;
+      break;
+    default:
+      LOG_ERROR("invalid access mode");
+      return false;
+  }
+
+  if (!LandmarkDatabase::open_database()) {
+    return false;
+  }
+
+  if (access_mode_ == AccessMode::ReadWriteCreate) {
+    if (!create_landmarks_table()) {
+      return false;
+    }
+    if (!create_spatial_index()) {
+      return false;
+    }
+  }
+
+  if (!prepare_insert_stmt()) {
+    return false;
+  }
+  if (!prepare_bounding_box_stmt()) {
+    return false;
+  }
+
+  return true;
+}
```

Comment:
tons of error checking overall, basically all of this stuff is fatal. just log error and throw. no need to return booling and checkign and the bubble the boolean out to a higher level. just the let the exception roll up the call stack